pdf/renderer.go

```diff
@@ -1002,6 +1015,45 @@ func (w *pdfPageWriter) embedImage(img image.Image, enc canvas.ImageEncoding) pd
 	return name
 }
 
+func (w *pdfPageWriter) embedJpeg(img canvas.JPEGImage) pdfName {
+	size := img.Bounds().Size()
+	dict := pdfDict{
+		"Type":    pdfName("XObject"),
+		"Subtype": pdfName("Image"),
+		"Width":   size.X,
+		"Height":  size.Y,
+
+		// "ColorSpace":       will be set below
+		"BitsPerComponent": 8, // bpc
+		// "Interpolate":      true,
+		"Filter": pdfFilterDCT, // f
+	}
+
+	switch img.ColorModel() {
+	case color.GrayModel:
+		dict["ColorSpace"] = pdfName("DeviceGray")
+	case color.YCbCrModel:
+		dict["ColorSpace"] = pdfName("DeviceRGB")
+	case color.CMYKModel:
+		dict["ColorSpace"] = pdfName("DeviceCMYK")
+		dict["Decode"] = pdfArray([]interface{}{1, 0, 1, 0, 1, 0, 1, 0})
+	default:
+		panic("image JPEG buffer has unsupported color space: " + fmt.Sprint(img.ColorModel()))
+	}
+
+	ref := w.pdf.writeObject(pdfStream{
```

Comment:
From hereon is duplicate code of `embedImage`, perhaps we can return a dict and bytes so that embedImage can either use the JPEG code or re-encode as has always done